test/app-tap/uri.test.lua

```diff
@@ -177,12 +177,21 @@ local function test_parse_ipv6_format_by_rfc2732(test)
 end
 
 local function test_format(test)
-    test:plan(14)
+    test:plan(16)
     local u = uri.parse("user:password@localhost")
     test:is(uri.format(u), "user@localhost", "password removed")
     test:is(uri.format(u, false), "user@localhost", "password removed")
     test:is(uri.format(u, true), "user:password@localhost", "password kept")
 
+    -- URI with IPv6
+    u = uri.parse('[2a00:1148:b0ba:2016:12bf:48ff:fe78:fd10]')
+    test:is(uri.format(u), "[2a00:1148:b0ba:2016:12bf:48ff:fe78:fd10]", "URI format")
+
+    -- URI with IPv6 URL
+    u = uri.parse("http://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:80/index.html")
+    test:is(uri.format(u), "http://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:80/index.html",
+            "URI format")
```

Comment:
Nit: Can we move it into its own test case too?

Reply:
created a separate test case